src/test/ui/nll/user-annotations/closure-substs.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(nll)]
+
+// Test that we enforce user-provided type annotations on closures.
+
+fn foo<'a>() {
+    |x: &'a i32| -> &'static i32 {
+        return x; //~ ERROR
+    };
+}
+
+fn bar<'a>() {
```

Comment:
Is this lifetime supposed to be used?

Reply:
No, the point was to test without the explicit `'a`. I'll remove it.

Reply:
Well, I guess I was testing  two distinct things. Perhaps I should add some more tests to tile the space.

Reply:
Done.